cookbooks/tomography_based_plate_motions/plugins/tomography_based_plate_motions.h

```diff
@@ -0,0 +1,503 @@
+/*
+  Copyright (C) 2014 - 2018 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_cookbooks_tomography_based_plate_motions_h
+#define _aspect_cookbooks_tomography_based_plate_motions_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/lateral_averaging.h>
+#include <aspect/material_model/grain_size.h>
+#include <aspect/simulator_access.h>
+#include <aspect/material_model/rheology/ascii_depth_profile.h>
+#include <array>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+
+    /**
+     * A material model that relies on compositional fields that stand for
+     * average grain sizes of a mineral phase and source terms for them that
+     * determine the grain size evolution in dependence of the strain rate,
+     * temperature, phase transitions, and the creep regime.
+     * This material model only works if a compositional field
+     * named 'grain_size' is present. In the diffusion
+     * creep regime, the viscosity depends on this grain size. We use the grain
+     * size evolution laws described in Behn et al., 2009: "Implications of grain
+     * size evolution on the seismic structure of the oceanic upper mantle", Earth
+     * Planet. Sci. Letters, 282, 178–189. Other material parameters are either
+     * prescribed similar to the 'simple' material model, or read from data files
+     * that were generated by the Perplex or Hefesto software. The material model
+     * is described in more detail in Dannberg, J., Z. Eilon, U. Faul,
+     * R. Gassmöller, P. Moulik, and R. Myhill (2017): "The importance of
+     * grain size to mantle dynamics and seismological observations",
+     * Geochem. Geophys. Geosyst., 18, 3034–3061, doi:10.1002/2017GC006944.,
+     * which is the canonical reference for this material model.
```

Comment:
I dont think this is an adequate description of this material model, it has changed a lot since description was written.